compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5139,8 +5139,9 @@ public void visit(BLangElvisExpr elvisExpr, AnalyzerData data) {
         BType lhsType = checkExpr(elvisExpr.lhsExpr, data);
         BType actualType = symTable.semanticError;
         if (lhsType != symTable.semanticError) {
-            if (lhsType.tag == TypeTags.UNION && lhsType.isNullable()) {
-                BUnionType unionType = (BUnionType) lhsType;
+            BType referencedType = Types.getReferredType(lhsType);
+            if (referencedType.tag == TypeTags.UNION && referencedType.isNullable()) {
```

Comment:
For clarification, is this check correct? According to the specification, in the second step, it is mentioned that if the static type of L does not include nil, then the static type of the result is the static type of L. Why do we perform a nullable check and throw an error as `operator.not.supported`?
In spec 
```
 The static type of the result of L ?: R is determined from the static types of L and R as follows:

    if the static type of L is nil, then the static type of the result is the static type of R;
    if the static type of L does not include nil, then the static type of the result is the static type of L;
    otherwise, the static type of the result is the union of the difference of the static type of L and (), and the static type of R.


```

Reply:
This is being tracked by https://github.com/ballerina-platform/ballerina-lang/issues/35025.